src/components/measureList/MeasureList.tsx

```diff
@@ -1,19 +1,150 @@
+  activeTab: number;
+}) {
   const history = useHistory();
+  const [searchCriteria, setSearchCriteria] = useState("");
+  const measureServiceApi = useMeasureServiceApi();
+
+  const handleClearClick = async (event) => {
+    setSearchCriteria("");
+    const data = await measureServiceApi.fetchMeasures(
+      props.activeTab === 0,
+      10,
+      0
+    );
+    setPageProps(data);
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    if (searchCriteria) {
+      const data =
+        await measureServiceApi.searchMeasuresByMeasureNameOrEcqmTitle(
+          props.activeTab === 0,
+          10,
+          0,
+          searchCriteria
+        );
+      setPageProps(data);
+    }
+  };
+  const setPageProps = (data) => {
+    if (data) {
+      const { content, totalPages, totalElements, numberOfElements, pageable } =
+        data;
+      props.setTotalPages(totalPages);
+      props.setTotalItems(totalElements);
+      props.setVisibleItems(numberOfElements);
+
+      props.setMeasureList(content);
+      props.setOffset(pageable.offset);
+      props.setInitialLoad(false);
+    }
+  };
+
+  const searchInputProps = {
+    startAdornment: (
+      <InputAdornment position="start">
+        <SearchIcon />
+      </InputAdornment>
+    ),
+    endAdornment: (
+      <IconButton
+        aria-label="Clear-Search"
+        sx={{
+          visibility: searchCriteria ? "visible" : "hidden",
+        }}
+        onClick={handleClearClick}
+      >
+        <ClearIcon />
+      </IconButton>
+    ),
+  };
+
   return (
     <div data-testid="measure-list">
       <div tw="flex flex-col">
         <div tw="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8">
           <div tw="py-2 align-middle inline-block min-w-full sm:px-6 lg:px-8">
             <div>
+              <form onSubmit={handleSubmit}>
+                <table
+                  style={{ marginLeft: 20, marginTop: 20, marginBottom: 20 }}
+                >
+                  <thead>
+                    <tr>
+                      <TextField
+                        onChange={(newValue) => {
+                          setSearchCriteria(newValue.target.value);
+                        }}
+                        id="searchMeasure"
+                        name="searchMeasure"
+                        placeholder="Search Measure"
+                        type="search"
+                        fullWidth
+                        data-testid="measure-search-input"
+                        label="Filter Measures"
+                        variant="outlined"
+                        defaultValue={searchCriteria}
+                        value={searchCriteria}
+                        inputProps={{
+                          "data-testid": "searchMeasure-input",
+                          "aria-required": "false",
+                          "aria-describedby": "searchMeasure-helper-text",
+                        }}
+                        InputProps={searchInputProps}
```

Comment:
Does having two `InputProps` cause any issues?

Reply:
@jkotanchik-SB inputProps and InputProps are different things, InputProps are for input field components (here the search icon and the clear icon) so we need both in this case. I don't see any issues so far.

Reply:
Oh, `i` vs `I`. Missed that entirely, thanks!